File: Programmers/42586.py
# 스택/큐 기능개발

from collections import deque
import math


def solution(progresses, speeds):
    queue = deque(progresses)
    s = deque(speeds)
    l = len(progresses)
    answer = []
    cnt = 0
    while queue:
        k = math.ceil((100 - queue[0]) / s[0])
        for i in range(l):
            queue[i] += s[i] * k

        while queue and queue[0] >= 100:
            queue.popleft()
            s.popleft()
            cnt += 1
            l -= 1
        answer.append(cnt)
        cnt = 0

    return answer

File: Programmers/42586.js
// 스택,큐 - 기능개발
function solution(progresses, speeds) {
    let ans = [];
    let cnt = 0;
    let i = 0;
    const len = progresses.length;
    while (i < len) {
        if (progresses[i] >= 100) {
            cnt++;
            i++;
            continue;
        }
        ans.push(cnt);
        cnt = 1;
        let x = progresses[i];
        const leftday = Math.ceil((100 - x) / speeds[i]);
        for (let j = i + 1; j < len; j++) {
            progresses[j] += speeds[j] * leftday;
        }
        i++;
    }
    ans.push(cnt);
    return ans.slice(1);
}